runtime/src/snapshot_utils.rs

```diff
@@ -649,64 +896,280 @@ pub fn bank_from_snapshot_archive<P>(
 where
     P: AsRef<Path> + std::marker::Sync,
 {
-    let unpack_dir = tempfile::Builder::new()
-        .prefix(TMP_SNAPSHOT_PREFIX)
-        .tempdir_in(snapshot_path)?;
+    do_bank_from_snapshot_archives::<_, &Path>(
+        account_paths,
+        frozen_account_pubkeys,
+        snapshot_path,
+        snapshot_tar,
+        None,
+        archive_format,
+        genesis_config,
+        debug_keys,
+        additional_builtins,
+        account_secondary_indexes,
+        accounts_db_caching_enabled,
+        limit_load_slot_count_from_snapshot,
+        shrink_ratio,
+        test_hash_calculation,
+    )
+}
+
+/// Rebuild a bank from full and incremental snapshot archives
+#[cfg(test)]
+#[allow(clippy::too_many_arguments)]
+fn bank_from_incremental_snapshot_archive<P, Q>(
+    account_paths: &[PathBuf],
+    frozen_account_pubkeys: &[Pubkey],
+    snapshot_path: &Path,
+    full_snapshot_archive_path: P,
+    incremental_snapshot_archive_path: Q,
```

Comment:
If we just passed this as an option to the existing `bank_from_snapshot_archive` could we avoid this function?

Reply:
Yep, can have just the one top-level function and not add a new one. Will do!

Reply:
Done!